From: Cezary Wilmanski <cezary.wilmanski@adtran.com>
Date: Sun, 07 Sep 2025 22:21:18 +0200
Subject: [PATCH] net: phy: correct led triggers deinitialization

Reset `phy->phy_num_led_triggers` to `0` in phy_led_triggers_unregister()
and check for NULL before freeing the triggers array.
This prevents stale counter values and makes the function safer when called in
corner cases (for example after no registration).

Signed-off-by: Cezary Wilmanski <cezary.wilmanski@adtran.com>
---
 drivers/net/phy/phy_led_triggers.c |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

--- a/drivers/net/phy/phy_led_triggers.c
+++ b/drivers/net/phy/phy_led_triggers.c
@@ -143,15 +143,19 @@ void phy_led_triggers_unregister(struct
 {
 	int i;
 
-	for (i = 0; i < phy->phy_num_led_triggers; i++)
-		phy_led_trigger_unregister(&phy->phy_led_triggers[i]);
-	kfree(phy->phy_led_triggers);
-	phy->phy_led_triggers = NULL;
+	if (phy->phy_led_triggers) {
+		for (i = 0; i < phy->phy_num_led_triggers; i++) 
+			phy_led_trigger_unregister(&phy->phy_led_triggers[i]);
+		kfree(phy->phy_led_triggers);
+		phy->phy_led_triggers = NULL;
+	}
 
 	if (phy->led_link_trigger) {
 		phy_led_trigger_unregister(phy->led_link_trigger);
 		kfree(phy->led_link_trigger);
 		phy->led_link_trigger = NULL;
 	}
+
+	phy->phy_num_led_triggers = 0;
 }
 EXPORT_SYMBOL_GPL(phy_led_triggers_unregister);
